### DISCUSSION TITLE
place_not_swipe removal action help

### DISCUSSION THREAD

**Discussion by jeverling** [MiczFlor/RPi-Jukebox-RFID#1694] [Q&A] [answered]
Hi all, I'm trying to use the `place_not_swipe` option. 

In `rfid.yaml`:
```
      place_not_swipe:
        enabled: true
        card_removal_action:
          quick_select: pause
```

It seems, that you have to use `alias` instead of `quick_select` now, right? https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1584/files

If I just try `alias: pause`, I get an error:

```
Exception in thread read_00CRemover:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/home/pi/RPi-Jukebox-RFID/src/jukebox/components/rfid/reader/__init__.py", line 48, in run
    self.timeout_action()
TypeError: call_ignore_errors() got an unexpected keyword argument 'note'
```

Any idea how I have to set this option so it works, and pause is triggered when removing the card?

**Comment by ghost** [comment]
You're right. The config file entry must look like this:
~~~
      place_not_swipe:
        enabled: true
        card_removal_action:
          alias: pause
~~~
Well spotted! That is a bug in the config script (New issue: #1695)

I just tested that in my environment and it works. Could you provide the log file entries leading up to this behavior? Which card reader are you using?

**Comment by ghost** [comment, marked as answer]
The second behavior is also a bug (#1696). For now please use this work-around:

In `src/jukebox/components/rpc_command_alias.py` please modify the `pause` entry too look like this:

~~~
    'pause': {
        'package': 'player',
        'plugin': 'ctrl',
        'method': 'pause'},
~~~

i.e. delete entries `note` and `ignore_card_removal_action`

**Reply by jeverling** [reply]
Thank you! Unfortunately I realized my card reader does not actually support `place_not_swipe`. It has a LED that stays green as long as the card remains on the reader, so I thought that would work, but it only submits the ID to the computer once.
I ordered a RDM6300 now and will update once I get it.

**Reply by ghost** [reply]
Out of curiosity, which USB Reader did you use? And where did you order your RDM6300? I am missing a good source to reference in the documentation. (BTW: the fix is implemented: #1697 :-)

**Reply by jeverling** [reply]
That was fast, thank you! :-)

I have this USB reader: https://www.ebay.de/itm/154635693138 (Neuftech USB RFID Reader)
I ordered one from _YARONGTECH_ before which looks and behaves exactly the same (apart from branding).

I ordered the RDM6300 from here: https://www.amazon.de/-/en/gp/product/B07DK78757/

**Reply by jeverling** [reply]
I can confirm that `place_not_swipe` works beautifully with the RDM6300 and the above changes.

**Comment by ghost** [comment]
Issue is fixed in [Release 3.1.1](https://github.com/MiczFlor/RPi-Jukebox-RFID/releases/tag/V3.1.1)
